tests/unit/test_util/test_blockUntilConditionMet.py

```diff
@@ -0,0 +1,78 @@
+# A part of NonVisual Desktop Access (NVDA)
+# This file is covered by the GNU General Public License.
+# See the file COPYING for more details.
+# Copyright (C) 2022 NV Access Limited.
+
+"""Unit tests for the blockUntilConditionMet submodule.
+"""
+
+import unittest
+from unittest.mock import patch
+
+from utils import blockUntilConditionMet as _moduleUnderTest
+
+
+class _FakeTimer():
+	"""
+	Simulate sleeping and getting the current time,
+	so that the module under test is not dependent on real world time.
+	"""
+	def __init__(self) -> None:
+		self._fakeTime: float = 0.0
+
+	def sleep(self, secs: float) -> None:
+		self._fakeTime += secs
+
+	def time(self):
+		return self._fakeTime
+
+
+class Test_blockUntilConditionMet(unittest.TestCase):
+	def setUp(self) -> None:
+		self._timer = _FakeTimer()
+
+	def test_condition_succeeds_before_timeout(self):
+		giveUpAfterSecs = 5
+		pollInterval = 1
+		startTime = self._timer.time()
+
+		def succeedJustBeforeTimeOut(currentTime: float):
+			return (currentTime - startTime) >= (giveUpAfterSecs - pollInterval)

```

Comment:
I think this could be clearer:

``` python
class _FakeTimer():
	def isComplete(self, *args, **kwargs):  # value from getValue is not relevant.
		return self._fakeTime > self._timeToComplete
...
		_moduleUnderTest.blockUntilConditionMet(
			...
			shouldStopEvaluator= self._timer.isComplete,
		)

```